packages/shopify-app-remix/src/server/shopify-app.ts

```diff
@@ -64,7 +65,11 @@ export function shopifyApp<
   }
 
   const params: BasicParams = {api, config, logger};
-  const oauth = new AuthStrategy<Config, Resources>(params);
+  const oauth = new AuthCodeFlowStrategy(params);
+  const authStrategy = new AuthStrategy<Config, Resources>({
+    ...params,
+    strategy: oauth,
+  });
```

Comment:
If I'm reading this code correctly this is a way to share common auth concerns across the 2 different strategies.  What else did you consider here? For example:

1) Extracting common functionality into utility functions that are imported are used
2) Using 3 classes: 1 super class which contains common functionality. 1 strategy class per strategy extends that base class.

E.g:

```ts
// Does common stuff, e.g:
class Strategy {
    constructor({api, config, logger}) {
        this.api = api;
        this.config = config;
        this.logger = logger;
    }

    respondToBotRequest(request) {}
    respondToOptionsRequest(request) {}
    respondToBounceUrl(request) {}
    respondToExitIFrameUrl(request) {}

    respondWithCorsHeaders(response) {}

    getTokenFromSessionStorage(shop) {}
    getContext(session, sessionToken) {}

    // etc
}


class AuthCodeStrategy extends Strategy {

    authenticate(request) {
        try {
            // Common
            this.respondToBotRequest(request) {}
            this.respondToOptionsRequest(request) {}
            this.respondToBounceUrl(request) {}
            this.respondToExitIFrameUrl(request) {}

            return this.authCodeFlow(request)
        } catch (error) {
            if (errorOrResponse instanceof Response) {
                this.respondWithCorsHeaders(errorOrResponse);
            }

            throw errorOrResponse
        }
    }

    authCodeFlow(request) {
        // Do auth code flow stuff
        // Throw a error or Response if needed.
        // return context
    }
}
```

I'm asking because this is where my mind went to first for the problem of sharing common functionality across 2 implementations.